Testscripts/Linux/perf_mariadb.sh

```diff
@@ -0,0 +1,285 @@
+		LogErr "Failed to download sysbench"
+		SetTestStateAborted
+		exit 0
+	fi
+
+	tar -xzvf ${SYSBENCH_VERSION}.tar.gz
+	if [ $? -gt 0 ]; then
+		LogErr "Failed to unzip sysbench"
+		SetTestStateAborted
+		exit 0
+	fi
+
+	LogMsg "Building sysbench"
+	pushd "${HOMEDIR}/sysbench-${SYSBENCH_VERSION}"
+	./autogen.sh && ./configure && make && make install
+	if [ $? -gt 0 ]; then
+		LogErr "Failed to build sysbench"
+		SetTestStateAborted
+		exit 0
+	fi
+	popd
+	export PATH="/usr/local/bin:${PATH}"
+	LogMsg "Sysbench installed successfully"
+}
+
+function create_data_dir () {
+	disks=$(ssh "${server}" ". $UTIL_FILE && get_AvailableDisks")
+	# We just use only one disk
+	for disk in ${disks}
+	do
+		Run_SSHCommand "${server}" "mkdir -p ${db_path}"
+		Run_SSHCommand "${server}" "yes | mkfs.ext4 /dev/${disk}"
+		Run_SSHCommand "${server}" "mount /dev/${disk} ${db_path}"
+		Run_SSHCommand "${server}" "cp -rf /var/lib/mysql/* ${db_path}"
+		Run_SSHCommand "${server}" "chown -R mysql:mysql ${db_path}"
+		Run_SSHCommand "${server}" "chmod 0755 -R ${db_path}"
+		# We also need to ensure that all the parent directories of the datadir upwards 
+		# have "x" (execute) permissions for all (user, group, and other)
+		# Refer to https://mariadb.com/kb/en/what-to-do-if-mariadb-doesnt-start/#cant-create-test-file
+		Run_SSHCommand "${server}" "chmod 0755 -R ${db_parent_path}"
+		break
+	done
+}
+
+function config_mariadb () {
+	case "$DISTRO_NAME" in
+		oracle|rhel|centos)
+			# Mariadb is not enabled by default
+			#Run_SSHCommand "${server}" "service mariadb start"
+			Run_SSHCommand "${server}" "service mariadb stop"
+			Run_SSHCommand "${server}" "echo datadir = ${db_path} | sudo tee --append /etc/my.cnf.d/mariadb-server.cnf"
+			Run_SSHCommand "${server}" "echo bind-address = 0.0.0.0 | sudo tee --append /etc/my.cnf.d/mariadb-server.cnf"
+			Run_SSHCommand "${server}" "echo max_connections = 1024 | sudo tee --append /etc/my.cnf.d/mariadb-server.cnf"
+			# Config the systemd service to set the open files limit as infinity
+			# Refer to https://mariadb.com/kb/en/systemd/
+			Run_SSHCommand "${server}" "echo LimitNOFILE=infinity | sudo tee --append /lib/systemd/system/mariadb.service"
+			Run_SSHCommand "${server}" "systemctl daemon-reload"
+			# Config selinux to set enforcing to permissive
+			# Refer to https://blogs.oracle.com/jsmyth/selinux-and-mysql
+			Run_SSHCommand "${server}" "setenforce 0"
+			Run_SSHCommand "${server}" "service mariadb start"
+			;;
+		ubuntu|debian)
+			Run_SSHCommand "${server}" "service mysql stop"
+			Run_SSHCommand "${server}" "sed -i '/datadir/c\datadir = ${escaped_path}' /etc/mysql/mariadb.conf.d/50-server.cnf"
+			Run_SSHCommand "${server}" "sed -i '/bind-address/c\bind-address = 0\.0\.0\.0' /etc/mysql/mariadb.conf.d/50-server.cnf"
+			Run_SSHCommand "${server}" "sed -i '/max_connections/c\max_connections = 1024' /etc/mysql/mariadb.conf.d/50-server.cnf"
+			Run_SSHCommand "${server}" "service mariadb start"
```

Comment:
I see that in both distro cases you have `Run_SSHCommand "${server}" "service mariadb start"` as last command. It will be helpful to take it out of switch case and then the $? after it would be more appropriate. What's your take on this?